CHANGELOG.md

```diff
@@ -5,6 +5,10 @@ This project adheres to [Semantic Versioning](http://semver.org/).
 
 ## [Unreleased][unreleased]
 
+## [v4.0.0] - 2022-03-14
```

Comment:
do we _**need**_ a major version upgrade? (fwiw you will also need to change package json with the version its not automated yet)

Reply:
I thought so too but @suhasdeshpande wanted a major version upgrade. Do you feel strongly @suhasdeshpande?

Reply:
I'm gonna say this PR doesn't need a major version bump. But if we use a feature that doesn't work with the old version a major bump would be required.